apps/prairielearn/src/pages/studentAssessmentInstance/studentAssessmentInstance.sql

```diff
@@ -7,6 +7,49 @@ WHERE
   ai.id = $assessment_instance_id;
 
 -- BLOCK select_instance_questions
+WITH
+  variant_max_submission_scores AS (
+    SELECT
+      v.id AS variant_id,
+      max(s.score) AS max_submission_score
+    FROM
+      instance_questions AS iq
+      JOIN variants AS v ON (v.instance_question_id = iq.id)
+      JOIN submissions AS s ON (s.variant_id = v.id)
+      JOIN assessment_questions AS aq ON (aq.id = iq.assessment_question_id)
+    WHERE
+      iq.assessment_instance_id = $assessment_instance_id
+      AND aq.deleted_at IS NULL
+      AND s.score IS NOT NULL
+    GROUP BY
+      v.id
+  ),
+  instance_question_variants AS (
+    SELECT
+      iq.id AS instance_question_id,
+      jsonb_agg(
+        jsonb_build_object(
+          'variant_id',
+          v.id,
+          'max_submission_score',
+          COALESCE(vmss.max_submission_score, 0)
+        )
+        ORDER BY
+          v.date
+      ) AS variants
+    FROM
+      instance_questions AS iq
+      JOIN assessment_questions AS aq ON (aq.id = iq.assessment_question_id)
+      JOIN variants AS v ON (v.instance_question_id = iq.id)
+      LEFT JOIN variant_max_submission_scores AS vmss ON (vmss.variant_id = v.id)
+    WHERE
+      iq.assessment_instance_id = $assessment_instance_id
+      AND aq.deleted_at IS NULL
+      AND NOT v.open
+      AND NOT v.broken
+    GROUP BY
+      iq.id
+  )
```

Comment:
Similar comments as before about aq.deleted_at.

```suggestion
WITH
  variant_max_submission_scores AS (
    SELECT
      v.id AS variant_id,
      max(s.score) AS max_submission_score
    FROM
      instance_questions AS iq
      JOIN variants AS v ON (v.instance_question_id = iq.id)
      JOIN submissions AS s ON (s.variant_id = v.id)
    WHERE
      iq.assessment_instance_id = $assessment_instance_id
      AND s.score IS NOT NULL
    GROUP BY
      v.id
  ),
  instance_question_variants AS (
    SELECT
      iq.id AS instance_question_id,
      jsonb_agg(
        jsonb_build_object(
          'variant_id',
          v.id,
          'max_submission_score',
          COALESCE(vmss.max_submission_score, 0)
        )
        ORDER BY
          v.date
      ) AS variants
    FROM
      instance_questions AS iq
      JOIN variants AS v ON (v.instance_question_id = iq.id)
      LEFT JOIN variant_max_submission_scores AS vmss ON (vmss.variant_id = v.id)
    WHERE
      iq.assessment_instance_id = $assessment_instance_id
      AND NOT v.open
      AND NOT v.broken
    GROUP BY
      iq.id
  )
```

Reply:
Applied in ca475996c